README.md
# Fap Instructor
This is just a nice little test branch

![Release](https://github.com/fapinstructor/fapinstructor-client/workflows/Release/badge.svg)

A fap/joi game where you follow the provided instructions while watching a stream of content that teases and torments you.

## Introduction

**TODO: Fill in this section.**

## Table Of Contents:

- [⚙️ Quick start](docs/quick-start.md)
- [🗄️ Project structure](docs/project-structure.md)
- [🎮 Game guide](docs/game-guide.md)
- [👁️ Style guide](docs/style-guide.md)
- [⌨️ Contributing](docs/contributing.md)

## License

[GNU GPLv2](https://choosealicense.com/licenses/gpl-2.0/)


public/CHANGELOG.md
# Changelog

All notable changes to this project will be documented here.

The overall project roadmap and progress can be viewed on our [Trello Board](https://trello.com/b/JAIcnXRf/fap-instructor)

## 2022-01-09

- Added local file system selector (only available in supported browsers)

## 2021-11-25

- Added support for handy stroke zones.

## 2021-10-22

- Fixed an issue where the game config wasn't being submitted.

## 2021-10-20

- Fixed handy devices that run v3 firmware.
- Fixed the unique subreddit validation errors.

## 2021-10-16

- Allow pasting multiple subreddits.

## 2021-09-30

- Complete overhaul of the config page.

## 2021-07-24

- Added redgifs and reddit gallery support
- Fixed and rewrote the final game logic (orgasm, denial, ruined)

## 2021-07-14

- Added Handy hardware support!
- Added new partner banner

## 2021-06-23

- Added button to view previous image
- Added hotkeys to the previous and next image buttons
- A ton of performance optimizations
- A variety of bug fixes

## 2021-05-05

- Added hotkeys to allow triggering actions via keyboard

## 2021-05-02

- Added ability to sort columns

## 2021-04-30

- Added ⭐ stars ⭐. You can now favourite any game and view them within the games table!
- Added "Average Game Length" column to game search

## 2021-04-14

- Added the ability for users to delete their profile

## 2021-02-22

- Added an FAQ page

## 2021-02-21

- Fixed the hands-off loop bug

## 2021-01-27

- Added the ability to hide the beat meter

## 2021-01-07

- Made more adjustments to a variety of tasks for smoother transitions
- Made grip adjustments occur more often and fixed a couple of bugs around them

## 2020-12-30

- Added metronome bar and ability to toggle sound
- Polish on a variety of tasks

## 2020-12-12

- Fixed issue where gifs weren't loading properly

## 2020-12-04

- Center images vertically for a better viewing experience
- Add prompt for when leaving an active game

## 2020-10-01

- Bookmarks now link back to the original post where the media came from
- Load times should be a thousand times faster
- Images are now preloaded to make slide transitions smoother
- The Game End page now has a nav bar
- Filtering videos, images, gifs should work flawlessly
- Duplicate images will not appear nearly as often because of a more expansive media library I pull from

## 2020-08-13

- Bug fixes
- Updated remaining packages

## 2020-08-10

- Bug fixes
- Updated packages

## 2020-08-04

- Fixed a bug for users who are unable to log in
- Lots of polish and code cleanup

## 2020-06-23

- Bug fixes
- Added a warmup period before the game kicks off

## 2020-06-22

- Performance improvements
- Some bug fixes
- Better bug tracking and monitoring

## 2020-06-09

- Added the ability to search for games
- Moved the previously played games and created games from the profile page into the games search page

## 2020-05-27

- Added a created games table under the profile page

## 2020-05-26

- Added a configure game option to every shared link
- Games can be made private or public when being shared. If they are public they will be searchable in the upcoming finder feature.
- Polish and tech debt

## 2020-05-22

- Polish and bug fixes
- Added "Play Game" button on Share Game dialog

## 2020-05-19

- Add a new share game button
- Add a tagging system
- Show previously played games on the profile page

## 2020-02-14

- Add a fullscreen button to the game page

## 2020-02-13

- Made Navbar responsive for mobile devices

## 2020-02-10

- Site navigation
- Change log
- Login Options
- Profile Page (Features to come)
- Add error message when failing to generate a game link

## 2020-01-17

- New game link generation. This will reduce the length of the URL making it easier to share.


src/audio.ts
import { getRandomInclusiveInteger } from "@/utils/math";
import CEI1 from "@/assets/audio/cei/1.mp3";
import CEI2 from "@/assets/audio/cei/2.mp3";
import CEI3 from "@/assets/audio/cei/3.mp3";
// denied
import Denied1 from "@/assets/audio/denied/1.mp3";
// faster
import Faster1 from "@/assets/audio/faster/1.mp3";
import Faster2 from "@/assets/audio/faster/2.mp3";
// moan
import Moan1 from "@/assets/audio/moan/1.mp3";
import Moan2 from "@/assets/audio/moan/2.mp3";
import Moan3 from "@/assets/audio/moan/3.mp3";
import Moan4 from "@/assets/audio/moan/4.mp3";
import Moan5 from "@/assets/audio/moan/5.mp3";
import Moan6 from "@/assets/audio/moan/6.mp3";
import Moan7 from "@/assets/audio/moan/7.mp3";
import Moan8 from "@/assets/audio/moan/8.mp3";
import Moan9 from "@/assets/audio/moan/9.mp3";
import Moan10 from "@/assets/audio/moan/10.mp3";
import Moan11 from "@/assets/audio/moan/11.mp3";
import Moan12 from "@/assets/audio/moan/12.mp3";
import Moan13 from "@/assets/audio/moan/13.mp3";
import Moan14 from "@/assets/audio/moan/14.mp3";
import Moan15 from "@/assets/audio/moan/15.mp3";
import Moan16 from "@/assets/audio/moan/16.mp3";
import Moan17 from "@/assets/audio/moan/17.mp3";
import Moan18 from "@/assets/audio/moan/18.mp3";
import Moan19 from "@/assets/audio/moan/19.mp3";
import Moan20 from "@/assets/audio/moan/20.mp3";
import Moan21 from "@/assets/audio/moan/21.mp3";
import Moan22 from "@/assets/audio/moan/22.mp3";
import Moan23 from "@/assets/audio/moan/23.mp3";
import Moan24 from "@/assets/audio/moan/24.mp3";
// orgasm
import Orgasm1 from "@/assets/audio/orgasm/1.mp3";
import Orgasm2 from "@/assets/audio/orgasm/2.mp3";
import Orgasm3 from "@/assets/audio/orgasm/3.mp3";
// ruined
import Ruined1 from "@/assets/audio/ruined/1.mp3";
import Ruined2 from "@/assets/audio/ruined/2.mp3";
import Ruined3 from "@/assets/audio/ruined/3.mp3";
// misc.
import CardShuffle from "@/assets/audio/cardshuffle.mp3";
import DontCumYet from "@/assets/audio/dontcumyet.mp3";
import Edge from "@/assets/audio/edge.mp3";
import HoldEdge from "@/assets/audio/holdedge.mp3";
import KeepStroking from "@/assets/audio/keep_stroking.mp3";
import LongMoan from "@/assets/audio/longMoan.mp3";
import Obey from "@/assets/audio/obey.mp3";
import RuinItForMe from "@/assets/audio/ruinitforme.mp3";
import SlapBalls from "@/assets/audio/slapballs.mp3";
import SqueezeBalls from "@/assets/audio/squeezeballs.mp3";
import StartStrokingAgain from "@/assets/audio/start_stroking_again.mp3";
import StartGame from "@/assets/audio/startgame.mp3";
import Tick from "@/assets/audio/tick.mp3";
import Tighter from "@/assets/audio/tighter.mp3";

const audioLibrary = {
  cei: [CEI1, CEI2, CEI3],
  denied: [Denied1],
  faster: [Faster1, Faster2],
  moan: [
    Moan1,
    Moan2,
    Moan3,
    Moan4,
    Moan5,
    Moan6,
    Moan7,
    Moan8,
    Moan9,
    Moan10,
    Moan11,
    Moan12,
    Moan13,
    Moan14,
    Moan15,
    Moan16,
    Moan17,
    Moan18,
    Moan19,
    Moan20,
    Moan21,
    Moan22,
    Moan23,
    Moan24,
  ],
  longMoan: [LongMoan],
  edge: [Edge],
  orgasm: [Orgasm1, Orgasm2, Orgasm3],
  ruin: [RuinItForMe],
  ruined: [Ruined1, Ruined2, Ruined3],
  dontCum: [DontCumYet],
  shuffle: [CardShuffle],
  holdEdge: [HoldEdge],
  keepStroking: [KeepStroking],
  startStroking: [StartStrokingAgain],
  tighter: [Tighter],
  obey: [Obey],
  slapBalls: [SlapBalls],
  squeezeBalls: [SqueezeBalls],
  startGame: [StartGame],
  tick: [Tick],
};

export type Audios = keyof typeof audioLibrary;

export default function getAudioUrl(key: Audios) {
  const variations = audioLibrary[key];
  const chosenVariation =
    variations[getRandomInclusiveInteger(0, variations.length - 1)];

  return chosenVariation;
}
